asteroid_filterbanks/pcen.py

```diff
@@ -0,0 +1,270 @@
+            + self.delta
+        ) ** one_over_root - self.delta ** one_over_root
+        return out.transpose(1, -1), hidden_state
+
+
+class PCEN(_PCEN):
+    """Per-Channel Energy Normalization as described in [1].
+
+    This applies a fixed or learnable normalization by an exponential moving average smoother and a compression.
+
+    Args:
+        alpha: The AGC strength (or gain normalization strength) (α in the paper).
+            Defaults to 0.96
+        delta: Bias added before compression (δ in the paper).
+            Defaults to 2.0
+        root: One over exponent applied for compression (r in the paper).
+            Defaults to 2.0
+        floor: Offset added to compression, prevents division by zero. (ϵ in the paper)
+            Defaults to 1e-6
+        smooth: Smoothing coefficient (s in the paper).
+            Defaults to 0.04
+        n_channels: Number of channels in the time frequency representation.
+            Defaults to 1
+        trainable: If True, the parameters (alpha, delta, root and smooth) are trainable. If False, the parameters are fixed.
+            Individual parameters can set to be fixed or trainable by passing a dictionary of booleans, with the key
+            matching the parameter name and the value being either True (trainable) or False (fixed).
+            i.e. ``{"alpha": False, "delta": True, "root": False, "smooth": True}``
+            Defaults to False
+        per_channel_smoothing: If True, each channel has it's own smoothing coefficient.
+            Defaults to False
+
+    Examples
+
+        >>> audio = torch.randn(10, 2, 16_000)
+        >>> fb = STFTFB(kernel_size=256, n_filters=256, stride=128)
+        >>> enc = Encoder(fb)
+        >>> tf_rep = enc(audio)
+        >>> mag_spec = transform.mag(tf_rep)
+        >>> pcen = PCEN(n_channels=2)
+        >>> energy = pcen(mag_spec)
+        >>> # or alternative with a mel-spectrogram
+        >>> mel = MelScale(n_filters=256, sample_rate=16_000)
+        >>> mel_spec = mel(audio)
+        >>> energy = pcen(mag_spec)
+
+    References
+        [1]: Wang, Y., et al. "Trainable Frontend For Robust and Far-Field Keyword Spotting”, arXiv e-prints, 2016.
+             https://arxiv.org/pdf/1607.05666.pdf
+
+    Influenced by leaf-audio's tensorflow implementation
+        https://github.com/google-research/leaf-audio/blob/7ead2f9fe65da14c693c566fe8259ccaaf14129d/leaf_audio/postprocessing.py
+
+    See license here
+        https://github.com/google-research/leaf-audio/blob/master/LICENSE
+    """
+
+    def forward(self, mag_spec: torch.Tensor) -> torch.Tensor:
+        """Computes the PCEN from magnitude spectrum representation.
+
+        Args:
+            mag_spec: A tensor containing an magnitude spectrum representation.
+
+        Shapes
+            >>> (batch, n_channels, freq, time) -> (batch, n_channels, freq, time)
+            >>> (batch, freq, time) -> (batch, 1, freq, time) # Assumed to be mono-channel
```

Comment:
It would be better if the unsqueeze is corrected back, to have an expectable behavior.